prometheus_client/exposition.py

```diff
@@ -307,20 +324,77 @@ def sample_line(samples):
             raise
 
         for suffix, lines in sorted(om_samples.items()):
-            output.append('# HELP {} {}\n'.format(openmetrics.escape_metric_name(metric.name + suffix),
+            output.append('# HELP {} {}\n'.format(openmetrics.escape_metric_name(metric.name + suffix, escaping),
                                                   metric.documentation.replace('\\', r'\\').replace('\n', r'\n')))
-            output.append(f'# TYPE {openmetrics.escape_metric_name(metric.name + suffix)} gauge\n')
+            output.append(f'# TYPE {openmetrics.escape_metric_name(metric.name + suffix, escaping)} gauge\n')
             output.extend(lines)
     return ''.join(output).encode('utf-8')
 
 
 def choose_encoder(accept_header: str) -> Tuple[Callable[[CollectorRegistry], bytes], str]:
+    # Python client library accepts a much narrower range of content-types than
+    # Prometheus does -- UTF-8 is only supported on OpenMetrics v1.0.0.
     accept_header = accept_header or ''
+    escaping = openmetrics.UNDERSCORES
     for accepted in accept_header.split(','):
         if accepted.split(';')[0].strip() == 'application/openmetrics-text':
-            return (openmetrics.generate_latest,
-                    openmetrics.CONTENT_TYPE_LATEST)
-    return generate_latest, CONTENT_TYPE_LATEST
+            toks = accepted.split(';')
+            version = _get_version(toks)
+            escaping = _get_escaping(toks)
+            # Only return an escaping header if we have a good version and
+            # mimetype.
+            if Version(version) >= Version('1.0.0'):
```

Comment:
If no `version` is specified, e.g. `Accept: application/openmetrics-text`, this is throwing an `packaging.version.InvalidVersion` due to version being an empty string.

We should have a tests for sure, but also what version would we want to use in this case? I guess LATEST for each should be fine?

Reply:
I believe the correct behavior is to always fall back to 0.0.4 rather than use the latest (https://github.com/prometheus/docs/blob/main/docs/instrumenting/content_negotiation.md#selection-of-format)

Python doesn't seem to support  open metrics 0.0.4 so I believe we have to fall back to prometheus plain there too.

Reply:
Added some test cases

Reply:
ok great, I'll keep OM falling back to OM